airbyte-webapp/src/components/connectorBuilder/Builder/StreamConfigView.tsx

```diff
@@ -23,14 +28,22 @@ interface StreamConfigViewProps {
 }
 
 export const StreamConfigView: React.FC<StreamConfigViewProps> = ({ streamNum }) => {
+  const streamPath = `streams[${streamNum}]`;
+  const streamFieldPath = (fieldPath: string) => `${streamPath}.${fieldPath}`;
+
   const { formatMessage } = useIntl();
-  const [field, , helpers] = useField<BuilderStream[]>("streams");
+  const [field, meta, helpers] = useField<BuilderStream[]>("streams");
+  const currentStreamErrors = meta.error?.[streamNum] as FormikErrors<BuilderStream>;
+  const hasSchemaErrors = Boolean(currentStreamErrors?.schema);
+  const hasConfigErrors = Boolean(
+    Object.keys(currentStreamErrors || {}).filter((errorKey) => errorKey !== "schema").length > 0
+  );
```

Comment:
One small issue with just looking at `meta.error` directly is that fields which have errors but are untouched cause the error indicator to show on the tab, but the error is. not shown on the field because it is untouched.

Could we use the `useBuilderErrors` custom hook instead to ignore untouched fields, so that the error indicator is only ever shown if the error is also being shown on the field?

Reply:
Ah, right, forgot about that. We can't use useBuilderErrors directly because it can't split out the `schema` bit so we can have separate error indicators for the configuration and the schema tabs. I could try to get that into the existing logic, but we already have the issue to refactor this to make it easier to work, I would rather like to do that first.

I reverted my attempt from yesterday and going to merge like that for now and open a follow-up PR.